### PR TITLE
fix: Generates correct join error for lobby. 

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: resources/prosody-plugins/mod_muc_lobby_rooms.lua [hunk 1]
```diff
@@ -380,10 +380,13 @@ process_host_module(main_muc_component_config, function(host_module, host)
         local invitee = event.stanza.attr.from;
         local affiliation = room:get_affiliation(invitee);
         if not affiliation or affiliation == 'none' then
-            local reply = st.error_reply(stanza, 'auth', 'registration-required'):up();
+            local reply = st.error_reply(stanza, 'auth', 'registration-required');
             reply.tags[1].attr.code = '407';
```
Comment (ibauersachs) [inline resources/prosody-plugins/mod_muc_lobby_rooms.lua:384]: Are you sure this is correct and still sets the 407 code on the right element?

Comment (damencho) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:384]: Hum, good point, I will double-check that.

Comment (damencho) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:384]: Yes, it is the correct element. `<error type='auth' code='407'>`

Comment (ibauersachs) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:384]: Ok. I have no idea about how that works and am still profoundly confused about that :up() thing.

**Review by ibauersachs — APPROVED** [review]

**Pull request merged**